hellgrid.gemspec

```diff
@@ -21,5 +21,6 @@ Gem::Specification.new do |s|
 
   s.add_runtime_dependency 'bundler', ['>= 1.11.0', '< 1.17']
 
-  s.add_development_dependency 'rspec', '~> 3.4', '>= 3.4.0'
+  s.add_development_dependency 'rspec', '~> 3.7.0'
+  s.add_development_dependency 'simplecov', '~> 0.15.1'
```

Comment:
🙋 I deliberately didn't do this to ensure that `SimpleCov` wasn't loaded before it was `require`d in `spec_helper.rb`.

Reply:
🤔 It would be loaded when you do `require 'simplecov'`, right? There shouldn't be autoload for things in the gemspec

Reply:
I've added a `Simplecov.start` in `bin/hellgrid` under the `bundle/setup` and it is not loaded:

```
./bin/hellgrid:11:in `<main>': uninitialized constant Simplecov (NameError)
Did you mean?  SimpleDelegator
```

Reply:
I think you meant `SimpleCov` (note the case).

Reply:
> There shouldn't be autoload for things in the gemspec (or bundler is doing something)

From memory, the `gemspec` line in the `Gemfile` loads any gems defined in the `gemspec`.

Reply:
Ahh.. still same thing:

```
./bin/hellgrid:11:in `<main>': uninitialized constant SimpleCov (NameError)
Did you mean?  SimpleDelegator
```

or if I remove the require in the spec_helper

```
An error occurred while loading ./spec/hellgrid/matrix_spec.rb.
Failure/Error: SimpleCov.start

NameError:
  uninitialized constant SimpleCov
  Did you mean?  SimpleDelegator
# ./spec/spec_helper.rb:1:in `<top (required)>'
# ./spec/hellgrid/matrix_spec.rb:1:in `require'
```

Reply:
https://github.com/bundler/bundler/issues/2018 tl&dr point: if you don't require your things manually, the gem will fail when installed with `gem install` without bundler, so -> bundler only modifies the path and you do the requires

Reply:
http://bundler.io/v1.16/guides/creating_gem.html#using-other-gems

Reply:
👍 Thanks for doing the research on this!